### PR TITLE
Add smlnj Maker for Standard ML

### PR DESCRIPTION
**Pull request by prikhi**

### CODE REVIEW

File: autoload/neomake/makers/ft/sml.vim [hunk 1]
```diff
@@ -0,0 +1,29 @@
+" vim: ts=4 sw=4 et
+
+function! neomake#makers#ft#sml#EnabledMakers()
```
Comment (benekastah) [inline autoload/neomake/makers/ft/sml.vim]: This function can simply be:

```
function! neomake#makers#ft#sml#EnabledMakers()
    return ['smlnj']
endfunction
```

Before we try to use a maker, `neomake#utils#Exists` is called on its `exe` property. We only need to check if things exist directly in functions like this when there are special requirements (e.g. only enable maker x if maker y is not present).

File: autoload/neomake/makers/ft/sml.vim [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    if neomake#utils#Exists('sml')
+        call add(makers, 'smlnj')
+    endif
+
+    let s:sml_makers = makers
+    return makers
+endfunction
+
+" This comes straight out of syntastic.
+function! neomake#makers#ft#sml#smlnj
```
Comment (benekastah) [inline autoload/neomake/makers/ft/sml.vim]: Any reason why this is `smlnj` instead of `sml`?

Comment (prikhi) [inline autoload/neomake/makers/ft/sml.vim]: There are many implementations of standard ml, the sml command is only provided by the SML/NJ implementation(which usually has a package name of `smlnj`).

File: autoload/neomake/makers/ft/sml.vim (hunk 1 above)
Comment (prikhi) [inline autoload/neomake/makers/ft/sml.vim]: OK will fix, just used the python file as a template.

**Comment by prikhi** [comment]
Fixed & rebased

**Comment by benekastah** [comment]
Thanks!

**Pull request merged**